dist/index.js

```diff
@@ -357,13 +508,63 @@ var Compiler = (function () {
             case RuleType.AnyOf:
                 return new TsTypes_1.TsType.Union(rule.anyOf.map(function (_) { return _this.toTsType(_); }));
             case RuleType.Reference:
-                return this.resolveType(rule.$ref);
+                return this.resolveType(rule.$ref, propName);
         }
         throw new Error('Unknown rule:' + rule.toString());
     };
-    Compiler.prototype.toTsType = function (rule) {
-        var type = this.createTsType(rule);
-        type.id = type.id || rule.id || rule.title;
+    Compiler.prototype.validateEnumMembers = function (rule) {
+        if (!rule.type)
+            rule.type = 'string';
+        var isDeclaredStringEnum = rule.type === 'string';
+        var isDeclaredIntegerEnum = rule.type === 'integer';
+        if (!isDeclaredStringEnum && !isDeclaredIntegerEnum) {
+            throw TypeError('Enum type must be string or integer; default is string if undefined');
+        }
+        if (rule.enum.some(function (_) { return _ instanceof Object; })) {
+            throw TypeError('Enum members must be a list of strings or a list of integers; instead, found an Object');
+        }
+        var isActuallyStringEnum = rule.enum.every(function (_) { return typeof (_) === 'string'; });
+        var isActuallyIntegerEnum = rule.enum.every(function (_) { return typeof (_) === 'number'; });
+        var isIntegerEnumWithValidStringValues = isActuallyIntegerEnum
+            && rule.tsEnumNames
+            && rule.tsEnumNames.length === rule.enum.length
+            && rule.tsEnumNames.every(function (_) { return typeof (_) === 'string'; });
+        if (isDeclaredStringEnum && !isActuallyStringEnum) {
+            throw TypeError('Enum was declared as a string type but found at least one non-string member');
+        }
+        if (isDeclaredIntegerEnum && !isIntegerEnumWithValidStringValues) {
+            if (!isActuallyIntegerEnum) {
+                throw TypeError('Enum was declared as an integer type, but found at least one non-integer member');
+            }
+            if (!rule.tsEnumNames) {
+                throw TypeError('Property tsEnumValues is required when enum is declared as an integer type');
```

Comment:
What are enumValues for? I thought we decided to only support enumNames. For future reference, from the chat:

| `enum`            | `enumNames`       | generated enum                     |
|-------------------|-------------------|------------------------------------|
| `[1, 2, 3]`       | `null`            | `enum {"1" = 1, "2" = 2, "3" = 3}` |
| `[1, 2, 3]`       | `["A", "B", "C"]` | `enum {A = 1, B = 2, C = 3}`       |
| `["A", "B", "C"]` | `null`            | `"A"|"B"|"C"`  |